low-level/sensors/impl/generic/raid.py

```diff
@@ -214,8 +222,12 @@ def _notify_NodeDataMsgHandler(self):
 
                 if self.alert_type is not None and self._drive_state_changed == True:
                     self._prev_drive_dict[device] = drive_dict[device]
-                    self._send_json_msg(self.alert_type, resource_id, device, self._drives[device])
-
+                    # Check RAID device fault state and send alert
+                    status = self._check_raid_fault_state(device, self.alert_type)
```

Comment:
for a specific RAIDed drive alert, checking raid darray device state? Here if one drive was with fault earlier and now that 
 first one recovered and second one has fault, which makes raid device also to be faulty, then these alerts wont be sent ? We need to persists at same granularity level (drive) as what logic is looking for. Or am i mis interpreting something